stix_shifter/stix_transmission/src/modules/utils/RestApiClient.py

```diff
@@ -53,8 +54,32 @@ def call_api(self, endpoint, method, headers=None, params=[], data=None, urldata
                 url = 'https://' + self.server_ip + '/' + endpoint
             try:
                 call = getattr(requests, method.lower())
-                response = call(url, headers=actual_headers, cert=self.cert_file_name, data=data, verify=self.cert_verify)
-                
+
+#                response = call(url, headers=actual_headers, cert=self.cert_file_name, data=data, verify=self.cert_verify)
+#  Subroto: Replaced the above line with the following because params is not passed to the call though the
+#  'call_api' signature has params as a parameter.
+#
+# This code is added because Guardium api requires 'params' to be sent to get the 
+# Authorization token AND 'data' to be sent to retrieve report information
+#
+                logging.debug(url)
+                if (not params or params is None):
```

Comment:
`not params` is the same as `params is None`.

Reply:
Please help: I am trying to distinguish between a) params=[] and b) params = None.  That's why I wrote (  if (not params or params is None): ).  Does  ("if no params") cover both the conditions?

Reply:
You just want `if (not params):` as it will cover both None and an empty list.